cin_validator/rules/cin2022_23/rule_1510.py

```diff
@@ -0,0 +1,100 @@
+    module=CINTable.ChildIdentifiers,
+    # replace the message with the corresponding value for this rule, gotten from the excel sheet.
+    message="UPN invalid (wrong check letter at character 1)",
+    # The column names tend to be the words within the < > signs in the github issue description.
+    affected_fields=[UPN],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    # Replace ChildIdentifiers with the name of the table you need.
+    df = data_container[ChildIdentifiers]
+
+    # implement rule logic as descriped by the Github issue. Put the description as a comment above the implementation as shown.
+    """
+    <UPN> (N00001) if present must contain the correct check letter
+
+    To calculate the check letter:
+
+    1. Multiply the individual digits by their weights as follows:
+
+    digit 2 by weight 2; digit 3 by weight 3; digit 4 by weight 4; digit 5 by weight 5; digit 6 by weight 6; digit 7 by weight 7; digit 8 by weight 8; digit 9 by weight 9; digit 10 by weight 10; digit 11 by weight 11; digit 12 by weight 12; digit 13 by weight 13.
+
+    2. Sum the individual results, divide the total by 23, and take the remainder.
+
+    3. Calculate the check letter from the result as follows:
+
+    0  = A;  1  = B;  2  = C;  3  = D;  4  = E;  5 = F;  6 = G;
+    7 = H;  8 = J;  9 = K;  10 = L;  11 = M;  12 = N;  13 = P;
+    14 = Q;  15 = R;  16 = T;  17 = U;  18 = V;  19 = W;  20 = X;
+    """
+    df.reset_index(inplace=True)
+    df2 = df[['index','UPN']]
+    df2 = df2[(df2['UPN'].str.len() == 13) & df2['UPN'].notna()]
+    df2['FIRST_CHAR'] = df2['UPN'].str[0]
+    df2['LAST_C'] = df2['UPN'].str[1: ]
+    df2['LAST_C'] = df2['LAST_C'].apply(lambda x: int(x) if str(x).isdigit() else pd.NA)
+    df2 = df2[df2['LAST_C'].notna()]
+    for i in range(1,13):
+        df2['C' + str(i)] = (df2['UPN'].str[i].astype(int) * (i+1)) % 23
+    df2['SUMMED'] = df2[['C1','C2','C3','C4','C5','C6','C7','C8','C9','C10','C11','C12']].sum(axis=1) % 23
+
+    check_map = enumerate(list('ABCDEFGHJKLMNPQRTUVWX'))
+    check_map = dict((i, j) for i, j in check_map)
+    df2['CHECK_CHAR'] = df2['SUMMED'].map(check_map)
+
+    df2 = df2[df2['CHECK_CHAR'].astype(str) != df2['FIRST_CHAR'].astype(str)]
+
+    failing_indices = df2.set_index('index').index
+
+    rule_context.push_issue(
+        table=ChildIdentifiers, field=UPN, row=failing_indices
+    )
+
+
+def test_validate():
+
+    child_identifiers = pd.DataFrame({"UPN": [
+    #These should pass
+    'A950000178301', #0 Valid format
+    pd.NA,           #1
+    'H243278544154', #2 Valid format
+    'ASFFAGSVSV123', #3 Nonsense
+    'R325',          #4 Nonsense   
+    #These should fail
+    'R247962919251', #5 Wrong initial char
+    'X428558133462'  #6 Wrong initial char
+    ]})
+
+    result = run_rule(validate, {ChildIdentifiers: child_identifiers})
+
+    issues = list(result.issues)
+
+    #assert len(issues) == 2
```

Comment:
Yes, you are correct. Two error positions will be flagged (index 5 and 6 as you pointed out below). 
You can uncomment this line.